File: server/src/users/user.model.ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { Type } from 'class-transformer';
import mongoose from 'mongoose';
import { Document } from 'mongoose';
import { Choice, ChoiceDocument, ChoiceSchema } from 'src/choices/choices.model';
import { Game } from 'src/games/game.model';

export type UserDocument = User & Document;

@Schema({
    toJSON: {
        getters: true,
        virtuals: true,
      }, 
      toObject:{
          getters: true,
          virtuals: true,
      }, 
  })
export class User {
  @Prop({ type: mongoose.Schema.Types.ObjectId, ref: 'Game',  required: true })
  @Type(() => Game)
  playingGameId: Game;

  @Prop({ required: true })
  nickname: string;

  @Prop({ required: true })
  isHost: boolean;

}
const UserSchema = SchemaFactory.createForClass(User);

export { UserSchema };

File: client/lib/models/user_model.dart
class User {
  String? id;
  String? nickname;
  String? playingGameId;
  int? pin;
  bool? isHost;

  User({this.id, this.nickname, this.playingGameId, this.pin, this.isHost});

  User.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    nickname = json['nickname'];
    playingGameId = json['playingGameId'];
    pin = json['pin'];
    isHost = json['isHost'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.id;
    data['nickname'] = this.nickname;
    data['playingGameId'] = this.playingGameId;
    data['pin'] = this.pin;
    data['isHost'] = this.isHost;
    data.removeWhere((key, value) => key == null || value == null);
    return data;
  }
}
